molpipeline/mol2mol/filter.py

```diff
@@ -120,38 +164,207 @@ def pretransform_single(self, value: RDKitMol) -> OptionalMol:
+    ) -> float:
+        """Calculate a single smiles match count for a molecule.
+
+        Parameters
+        ----------
+        filter_element: Any
+            smiles to calculate match count for.
+        value: RDKitMol
+            Molecule to calculate smiles match count for.
+
+        Returns
+        -------
+        float
+            smiles match count value.
+        """
+        return len(value.GetSubstructMatches(Chem.MolFromSmiles(filter_element)))
+
+
+class DescriptorsFilter(_BaseKeepMatchesFilter):
+    """Filter to keep or remove molecules based on RDKit descriptors."""
+
+    def __init__(
         self,
-        name: str = "MixtureFilter",
+        descriptors: dict[str, tuple[Optional[float], Optional[float]]],
+        keep_matches: bool = True,
+        mode: Literal["any", "all"] = "any",
+        name: Optional[str] = None,
         n_jobs: int = 1,
         uuid: Optional[str] = None,
     ) -> None:
-        """Initialize MixtureFilter.
+        """Initialize DescriptorsFilter.
 
         Parameters
         ----------
-        name: str, optional (default: "MixtureFilterPipe")
+        descriptors: dict[str, tuple[Optional[float], Optional[float]]]
+            Dictionary of RDKit descriptors to filter by.
+            The value must be a tuple of minimum and maximum. If None, no limit is set.
+        keep_matches: bool, optional (default: True)
+            If True, molecules containing the specified descriptors are kept, else removed.
+        mode: Literal["any", "all"], optional (default: "any")
+            If "any", at least one of the specified descriptors must be present in the molecule.
+            If "all", all of the specified descriptors must be present in the molecule.
+        name: Optional[str], optional (default: None)
             Name of the pipeline element.
         n_jobs: int, optional (default: 1)
             Number of parallel jobs to use.
         uuid: str, optional (default: None)
             Unique identifier of the pipeline element.
         """
-        super().__init__(name=name, n_jobs=n_jobs, uuid=uuid)
+        super().__init__(
+            keep_matches=keep_matches, mode=mode, name=name, n_jobs=n_jobs, uuid=uuid
+        )
+        self.descriptors = descriptors
+
+    @property
+    def descriptors(self) -> dict[str, tuple[Optional[float], Optional[float]]]:
+        """Get allowed descriptors as dict."""
+        return self._descriptors
+
+    @descriptors.setter
+    def descriptors(
+        self, descriptors: dict[str, tuple[Optional[float], Optional[float]]]
+    ) -> None:
+        """Set allowed descriptors as dict.
+
+        Parameters
+        ----------
+        descriptors: dict[str, tuple[Optional[float], Optional[float]]]
+            Dictionary of RDKit descriptors to filter by.
+        """
+        self._descriptors = descriptors
+        if not all(hasattr(Descriptors, descriptor) for descriptor in descriptors):
```

Comment:
I would rather call the class `RDKitDescriptorsFilter` instead of `DescriptorsFilter` because it specifically works with RDKit's `Descriptors` module